Create new user regions

oskarhurst opened the pull request:
This Pr is to add a new table to to the data base which hold User defined custom geoJson regions. These regions can then be use to get a bounding box used in the Product downloads.

Commit:
Create new user regions

MikeNeilson commented on sql/common/V2.62.0__user_saved_regions.sql:
```diff
@@ -0,0 +1,103 @@
+CREATE INDEX idx_user_region_tags ON user_region USING GIN(tags);
+
+-- Function to validate GeoJSON before saving
+CREATE OR REPLACE FUNCTION validate_user_region_geojson()
```
Create a temporary empty trigger function here and move the actual trigger to a repeatable migration.

Makes it easier to update the trigger itself later.

MikeNeilson commented on sql/common/V2.62.0__user_saved_regions.sql:
```diff
@@ -0,0 +1,103 @@
+    EXECUTE FUNCTION validate_user_region_geojson();
+
+-- View for user regions with additional computed fields
+CREATE OR REPLACE VIEW v_user_region AS
```
Put views in their own repeatable migration.

If something needs the view to exist for reference, like say a function or something you can define the view by name and the columns but leave it all as placeholders.

MikeNeilson commented:
The API logic itself looks fine, you did find several database migration traps that need to be avoided.

Commit:
add repeatables

oskarhurst commented:
@MikeNeilson I didn't reference the view or trigger any where else in the version file so i just removed and added them to repeatable file. Is there any other concerns?

MikeNeilson commented on sql/common/R__08_functions_user_regions.sql:
```diff
@@ -0,0 +1,40 @@
+$$ LANGUAGE plpgsql;
+
+-- Add trigger for validation
+CREATE TRIGGER validate_user_region_before_insert_update
```
```suggestion
CREATE OR REPLACE TRIGGER validate_user_region_before_insert_update
```

Though I usually put these in afterMigrate scripts. Not a requirement, really depends on the overall organization used.

oskarhurst committed:
Update sql/common/R__08_functions_user_regions.sql

Co-authored-by: Mike Neilson <gamingmike@gmail.com>
